ntoskrnl/io/iomgr/deviface.c

```diff
@@ -304,7 +305,189 @@ IoGetDeviceInterfaceAlias(IN PUNICODE_STRING SymbolicLinkName,
                           IN CONST GUID *AliasInterfaceClassGuid,
                           OUT PUNICODE_STRING AliasSymbolicLinkName)
 {
-    return STATUS_NOT_IMPLEMENTED;
+    UNICODE_STRING BaseKeyName = RTL_CONSTANT_STRING(L"\\Registry\\Machine\\System\\CurrentControlSet\\Control\\Class\\");
+    UNICODE_STRING AliasKeyName = RTL_CONSTANT_STRING(L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\");
+    UNICODE_STRING bipName, InLinkName;
+    UNICODE_STRING GuidString;
+    HANDLE InterfaceKey = NULL;
+    PKEY_FULL_INFORMATION fip = NULL;
+    PKEY_BASIC_INFORMATION bip = NULL;
+    PWCHAR pdest;
+    ULONG i, Size;
+    LONG ret;
+    NTSTATUS Status;
+
+    PAGED_CODE();
+
```

Comment:
Missing: Sanity checks: What happens if SymbolicLinkName is empty?

Reply:
Added sanity check for `SymbolicLinkName`.